dask/dataframe/core.py

```diff
@@ -3946,11 +3946,13 @@ def quantile(df, q):
     q : list/array of floats
         Iterable of numbers ranging from 0 to 100 for the desired quantiles
     """
-    # current implementation needs q to be sorted, sort to make sure
-    q = np.asanyarray(q)
-    if q.ndim > 0:
-        q = np.sort(q, kind='mergesort')
-        
+    # current implementation needs q to be sorted so 
+    # sort if array-like, otherwise leave it alone
+    q_ndarray = np.array(q)
+    if q_ndarray.ndim > 0:
```

Comment:
In this case, maybe want to test for the single-value/scalar case?

Reply:
Is there a straightforward way to tell if an arg is a scalar in python? Maybe there is now that there are base classes everywhere. It used to be hard so I'm used to doing the thing where I try to convert it to an array with np.array, then check to see if it has non-zero dimension, sorting if it does then setting q=q_ndarray and simply abandoning q_ndarray if q_ndarray.ndim==0.

By the way, I'm off on vacation in a couple hours. Feel free to modify and commit in my absence. I don't care who GIT blames for this.